test/unit/controllers/DeveloperControllerSpec.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers
+
+import connectors.AuthConnector.InvalidCredentials
+import connectors.{ApplicationConnector, AuthConnector, DeveloperConnector}
+import controllers.DevelopersController
+import model.LoginDetails.{JsonStringDecryption, JsonStringEncryption}
+import model._
+import org.joda.time.DateTime
+import org.mockito.ArgumentCaptor
+import org.mockito.BDDMockito._
+import org.mockito.Matchers._
+import org.scalatest.mock.MockitoSugar
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import play.api.test.{FakeRequest, Helpers}
+import play.filters.csrf.CSRF.SignedTokenProvider
+import uk.gov.hmrc.crypto.Protected
+import uk.gov.hmrc.play.frontend.auth.AuthenticationProvider
+import uk.gov.hmrc.play.http.{HeaderCarrier, SessionKeys}
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+
+import scala.concurrent.Future
+
+class DeveloperControllerSpec extends UnitSpec with MockitoSugar with WithFakeApplication {
+
+  Helpers.running(fakeApplication) {
+
+    trait Setup {
+      val underTest = new DevelopersController {
+        val authConnector = mock[AuthConnector]
+        val authProvider = mock[AuthenticationProvider]
+        val applicationConnector = mock[ApplicationConnector]
+        val developerConnector = mock[DeveloperConnector]
+      }
```

Comment:
Nevermind - I saw that the `GatewayAuthWrapper` requires these other mocks to function correctly.